pricelist_item_generator/models/pricelist.py

```diff
@@ -0,0 +1,415 @@
+    def activate_generator(self):
+        for generator in self:
+            if generator.active is False:
+                vals = {'active': True, 'to_update': True}
+            else:
+                vals = {'active': False}
+                items_tpl = self.env['pricelist.item.template'].search(
+                    [('price_generator_id', '=', generator.id)])
+                if items_tpl:
+                    items2unlink = self.env['product.pricelist.item'].search([
+                        ('item_template_id', 'in', [x.id for x in items_tpl])])
+                    if items2unlink:
+                        items2unlink.unlink()
+                # 'todo' field to initial value
+                generator.item_template_ids.write({'todo': 'create'})
+                generator.product_condition_ids.write({'todo': 'create'})
+            generator.write(vals)
+            return True
+
+
+class PricelistProductCondition(models.Model):
+    _name = "pricelist.product.condition"
+    _description = "Products selection by criterias"
+    _rec_name = 'id'
+
+    product_id = fields.Many2one(
+        comodel_name='product.product', string='Product Variant',
+        help="If product is selected, no other criterias can be take account")
+    product_tmpl_id = fields.Many2one(
+        comodel_name='product.template', string='Product',
+        help="If product is selected, no other criterias can be take account")
+    categ_id = fields.Many2one(
+        comodel_name='product.category', string='Category',
+        help="Products of the category or products of children categories")
+    price_generator_id = fields.Many2one(
+        comodel_name='pricelist.item.generator',
+        string='Generator', required=True)
+    todo = fields.Selection(
+        selection=TODO_SELECTION, readonly=True, string='Next Action',
+        help="Required action on Pricelist Items")
+
+    @api.model
+    def create(self, vals):
+        self.env['pricelist.item.generator'].search(
+            [('id', '=', vals['price_generator_id'])]).write(
+            {'to_update': True})
+        vals['todo'] = 'create'
+        return super(PricelistProductCondition, self).create(vals)
+
+    @api.multi
+    def write(self, vals):
+        # Some records shouldn't be written with the same values
+        todo_update = update_item_tpl_or_condition(vals, self)
+        super(PricelistProductCondition, self).write(vals)
+        if todo_update:
+            super(PricelistProductCondition, todo_update).write(
+                {'todo': 'update'})
+        return True
+
+
+class AbstractPriceListItemGenerator(models.AbstractModel):
+    _name = "abstract.pricelist.item.generator"
+    _description = "Abstract Pricelist Items Generator"
+
+    price_generator_id = fields.Many2one(
+        comodel_name='pricelist.item.generator',
+        string='Generator', readonly=True)
+    min_quantity = fields.Integer(default=1)
+
+
+class PricelistItemTemplateBase(models.Model):
```

Comment:
Class names should be the UpperCamelCaseForm of the model name, so in this instance it should be `PricelistItemTemplate`